model/atmosphere/dycore/src/icon4py/model/atmosphere/dycore/stencils/compute_edge_diagnostics_for_dycore_and_update_vn.py

```diff
@@ -213,28 +195,25 @@ def _compute_theta_rho_face_values_and_pressure_gradient_and_update_vn(
         horizontal_pressure_gradient,
     )
 
-    horizontal_pressure_gradient = (
-        concat_where(
-            (nflatlev <= dims.KDim) & (dims.KDim < (nflat_gradp + 1)),
-            concat_where(
-                (start_edge_nudging_level_2 <= dims.EdgeDim) & (dims.EdgeDim < end_edge_local),
-                _compute_horizontal_gradient_of_exner_pressure_for_nonflat_coordinates(
-                    inv_dual_edge_length=inv_dual_edge_length,
-                    z_exner_ex_pr=temporal_extrapolation_of_perturbed_exner,
-                    ddxn_z_full=ddxn_z_full,
-                    c_lin_e=c_lin_e,
-                    z_dexner_dz_c_1=ddz_of_temporal_extrapolation_of_perturbed_exner_on_model_levels,
+    if igradp_method == horzpres_discr_type.TAYLOR_HYDRO:
+        if nflatlev < (nflat_gradp + 1):  # TODO check this condition
```

Comment:
Is this condition needed? otherwise we could put the 3 k intervals in a nice switch like thing instead